lib/provider.ex

```diff
@@ -108,13 +108,13 @@ defmodule Provider do
   # ------------------------------------------------------------------------
 
   @doc "Retrieves all params according to the given specification."
-  @spec fetch_all(source, params) :: {:ok, data} | {:error, [String.t()]}
-  def fetch_all(source, params) do
+  @spec fetch_all(source, params, Keyword.t()) :: {:ok, data} | {:error, [String.t()]}
+  def fetch_all(source, params, opts) do
```

Comment:
These opts are passed to the source, right? It would be good to document that somewhere.